solr/modules/llm/src/java/org/apache/solr/llm/texttovector/update/processor/TextToVectorUpdateProcessor.java

```diff
@@ -0,0 +1,100 @@
+ */
+
+package org.apache.solr.llm.texttovector.update.processor;
+
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.SolrInputField;
+import org.apache.solr.llm.texttovector.model.SolrTextToVectorModel;
+import org.apache.solr.llm.texttovector.store.rest.ManagedTextToVectorModelStore;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.update.AddUpdateCommand;
+import org.apache.solr.update.processor.UpdateRequestProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
+
+
+class TextToVectorUpdateProcessor extends UpdateRequestProcessor {
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final String inputField;
+    private final String outputField;
+    private final String model;
+    private SolrTextToVectorModel textToVector;
+    private ManagedTextToVectorModelStore modelStore = null;
+
+    public TextToVectorUpdateProcessor(
+            String inputField,
+            String outputField,
+            String model,
+            SolrQueryRequest req,
+            UpdateRequestProcessor next) {
+        super(next);
+        this.inputField = inputField;
+        this.outputField = outputField;
+        this.model = model;
+        this.modelStore = ManagedTextToVectorModelStore.getManagedModelStore(req.getCore());
+    }
+
+    /**
+     * @param cmd the update command in input containing the Document to process
+     * @throws IOException If there is a low-level I/O error
+     */
+    @Override
+    public void processAdd(AddUpdateCommand cmd) throws IOException {
+        this.textToVector = modelStore.getModel(model);
+        if (textToVector == null) {
+            throw new SolrException(
+                    SolrException.ErrorCode.BAD_REQUEST,
+                    "The model requested '"
+                            + model
+                            + "' can't be found in the store: "
+                            + ManagedTextToVectorModelStore.REST_END_POINT);
+        }
+
+        SolrInputDocument doc = cmd.getSolrInputDocument();
+        SolrInputField inputFieldContent = doc.get(inputField);
+        if (!isNullOrEmpty(inputFieldContent, doc, inputField)) {
+            String textToVectorise = inputFieldContent.getValue().toString();//add null checks and
+            float[] vector = textToVector.vectorise(textToVectorise);
+            List<Float> vectorAsList = new ArrayList<Float>(vector.length);
+            for (float f : vector) {
+                vectorAsList.add(f);
+            }
+            doc.addField(outputField, vectorAsList);
+        }
+        super.processAdd(cmd);
+    }
+
+    protected boolean isNullOrEmpty(SolrInputField inputFieldContent, SolrInputDocument doc, String fieldName) {
```

Comment:
do we warn on these types of "data model" issues in other aspects of solr?  I mean, who would ever check solr logs to spot that these input fields are missing?   I see this cluttering the logs and not being very useful to have operationally..   I think I would just query solr "show me all docs that don't have the field populate" to find the issue....

Reply:
mmm I see your point, better if we just log a warning say "vectorisation failed", with the reason "null or empty source field" ?


I suspect that silent failure would be equally problematic to understand why there are no vectors? (I also just discovered that the 'vectorise' method could throw runtime exception)

Reply:
addressed, resolving this, feel free to open a new comment if necessary